tdrs-backend/tdpservice/users/models.py

```diff
@@ -38,10 +39,14 @@ def __str__(self):
         """Return the username as the string representation of the object."""
         return self.username
 
-    @property
-    def is_admin(self):
-        """Check if the user is an admin."""
-        return (
-            self.is_superuser
-            or Group.objects.get(name="OFA Admin") in self.groups.all()
-        )
+    def is_in_group(self, group_name: str) -> bool:
+        """Return whether or not the user is a member of the specified Group."""
+        return self.groups.filter(name=group_name).exists()
+
+    @cached_property
+    def is_data_analyst(self) -> bool:
```

Comment:
This can now be referenced from any user instance via `user.is_data_analyst`. This prevents needing to make function calls in multiple places and by using a `cached_property` here Django will store the results in memory until the model instance is de-allocated, which saves database calls in one of the permission classes which refers to this multiple times in a single thread.